src/Sarif.Viewer.VisualStudio/ResultTextMarker.cs

```diff
@@ -47,12 +57,63 @@ public class ResultTextMarker
         /// </summary>
         private bool? regionIsFullyPopulated;
 
-        private int runIndex;
-        private ISarifLocationTag tag;
+        /// <summary>
+        /// The persistent span created for this marker.
+        /// </summary>
+        /// <remarks>
+        /// Visual Studio's persistent span factory allows us to correctly track spans for documents that the users opens, edits and closes
+        /// and then re-opens. This allows for our "tags" to be accurate after an edit.
+        /// That is as long as they don't modify the document in another editor (say notepad).
+        /// </remarks>
+        IPersistentSpan persistentSpan;
 
         public string FullFilePath { get; set; }
```

Comment:
>public string FullFilePath { get; set; } [](start = 8, length = 40)

This is settable (ugh). We need to throw away the persistent span (if there is any) and send a property change.

Reply:
And comment what is happening.

---
In reply to: [502079843](https://github.com/microsoft/sarif-visualstudio-extension/pull/214#discussion_r502079843) [](ancestors = 502079843)